crates/lib/src/qpu/api.rs

```diff
@@ -260,11 +269,25 @@ impl ExecutionOptions {
         }
     }
 
-    async fn get_controller_client(
+    /// Get a controller client for the given QPU ID.
+    pub async fn get_controller_client(
         &self,
         client: &Qcs,
         quantum_processor_id: Option<&str>,
     ) -> Result<ControllerClient<RefreshService<Channel, ClientConfiguration>>, QpuApiError> {
+        let service = self
+            .get_qpu_grpc_connection(client, quantum_processor_id)
+            .await?;
+        Ok(ControllerClient::new(service)
+            .max_decoding_message_size(MAX_DECODING_MESSAGE_SIZE_BYTES))
+    }
+
+    /// Get a GRPC connection to a QPU, without specifying the API to use.
+    pub async fn get_qpu_grpc_connection(
+        &self,
+        client: &Qcs,
+        quantum_processor_id: Option<&str>,
+    ) -> Result<RefreshService<Channel, ClientConfiguration>, QpuApiError> {
```

Comment:
This is the heart of the change right, to add a public function that returns the wrapped `Channel`? Seems fine to me!

Reply:
Yep!